topostats/filters.py

```diff
@@ -1,128 +1,342 @@
-from statistics import median
+"""Contains filter functions that take a 2D array representing an image as an input, as well as necessary parameters,
+and return a 2D array of the same size representing the filtered image."""
+# pylint: disable=no-name-in-module
+# pylint: disable=invalid-name
+# pylint: disable=fixme
+from pathlib import Path
+from typing import Union
+import logging
 import numpy as np
+import pySPM
```

Comment:
I think this is a good idea and it should probably sit under `io`. It might perhaps be worthy of its own issue to address though so that an appropriate design pattern can be used to make the process flexible and open to extension to accommodate other file types.

Reply:
Please can you raise such an issue post merge @ns-rse?